### ISSUE TITLE
Divide operator precision

### ISSUE THREAD

**Issue by knutheless** [onepub-dev/money.dart#63]
I trying to calculate the base value of a price with VAT included.
Assuming a VAT of 21%, if I have a price of 121, the result of 121 / 1.21 must be 100, currently is 100.43
The problems seems to be in the divide function of Fixed, beeing the multiplier always with a scale of 2:

  Fixed divide(num divisor) {
    return this * Fixed.fromNum(1.0 / divisor.toDouble());
  }

**Comment by bsutton** [comment]
What happens if you change the scale of fixed:

```
return this * Fixed.fromNum(1.0 / divisor.toDouble(), scale: 4);
```

**Comment by bsutton** [comment]
I'm thinking that the correct solution is for Fixed to convert any double operand to a high scale fixed value before doing the operation.

For java at least, apparently you can calculate the number of decimals by taking the log of the number.
However it might just be easier to use the max which appears to be 16 digits of scale.
I've not tested it but I think this method would deliver the results that user 'expected' when the use a double with a fixed.
There might be some degradation in performance but I think for the Dart domain this isn't an issue.

Thoughts?

**Comment by bsutton** [comment]
I've released 2.0.0-beta.1

Fixed.fromNum now defaults to a scale of 16 and the above code now works as expected.
